server/src/main/java/io/crate/planner/node/ddl/AlterTablePlan.java

```diff
@@ -129,7 +137,20 @@ public static TableParameter getTableParameter(AlterTable<Object> node, TablePar
         return tableParameter;
     }
 
-    // Only check for permission if statement is not changing the metadata blocks, so don't block `re-enabling` these.
+    static void validateTableParametersForReplicatedTables(TableInfo tableInfo, Settings tableSettings) {
+        var subscriptionName = REPLICATION_SUBSCRIPTION_NAME.get(tableInfo.parameters());
+        var isSubscribed = subscriptionName != null && subscriptionName.isEmpty() == false;
+        if (isSubscribed) {
+            if (!SUPPORTED_SETTINGS_FOR_REPLICATED_TABLES.containsAll(tableSettings.keySet())) {
+                throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                "The relation \"%s\" doesn't support changing the setting, because it is included in a logical replication subscription.", tableInfo.ident().toString()));
+            }
+        }
+    }
+
+    private final static Set<String> SUPPORTED_SETTINGS_FOR_REPLICATED_TABLES = TableParameters
```

Comment:
Can't we use an instance of `TableParameters` containing this new set of settings instead? And switch it inside `getTableParameterInfo` similar to how we differentiate table vs partition supported settings. This would also eliminate the use of a dedicated validation method.